pkg/client/client.go

```diff
@@ -33,44 +33,81 @@ func NewApplyClient(cl client.Client, scheme *runtime.Scheme) *ApplyClient {
 	return &ApplyClient{cl: cl, scheme: scheme}
 }
 
+type createOrUpdateObjectConfiguration struct {
+	owner             v1.Object
+	forceUpdate       bool
+	saveConfiguration bool
+}
+
+func newCreateOrUpdateObjectConfiguration(options ...CreateOrUpdateObjectOption) createOrUpdateObjectConfiguration {
```

Comment:
I guess that we could make the names shorter - something like `applyObjectConfig`, but it's not critical

Reply:
good idea! Let me rename `createOrUpdateObjectConfiguration` to `applyObjectConfiguration` (I tend to prefer full words)